src/test/java/com/rackspace/ceres/app/services/MetadataServiceCachingTest.java

```diff
@@ -0,0 +1,187 @@
+
+  @MockBean
+  ReactiveStringRedisTemplate redisTemplate;
+
+  @Mock
+  ReactiveValueOperations<String, String> opsForValue;
+
+  @Autowired
+  MetadataService metadataService;
+
+  @Test
+  void writeThruToCassandraAtFirst() {
+    when(cassandraTemplate.insert(any(Object.class)))
+        .thenReturn(Mono.empty());
+
+    when(redisTemplate.opsForValue())
+        .thenReturn(opsForValue);
+    when(opsForValue.setIfAbsent(any(), any()))
+        .thenReturn(Mono.just(true));
+
+    final String tenant = randomAlphanumeric(10);
+    final String seriesSetHash = randomAlphanumeric(10);
+    final String metricName = randomAlphanumeric(10);
+    final String tagK = randomAlphanumeric(5);
+    final String tagV = randomAlphanumeric(5);
+    final Map<String, String> tags = Map.of(tagK, tagV);
+
+    Mono.from(
+        metadataService.storeMetadata(tenant, seriesSetHash, metricName, tags)
+    ).block();
+
+    verify(cassandraTemplate).insert(
+        new SeriesSetHash()
+            .setTenant(tenant)
+            .setSeriesSetHash(seriesSetHash)
+            .setMetricName(metricName)
+            .setTags(tags)
+    );
+    verify(cassandraTemplate).insert(
+        new MetricName()
+            .setTenant(tenant)
+            .setMetricName(metricName)
+    );
+    verify(cassandraTemplate).insert(
+        new SeriesSet()
+            .setTenant(tenant)
+            .setMetricName(metricName)
+            .setTagKey(tagK)
+            .setTagValue(tagV)
+            .setSeriesSetHash(seriesSetHash)
+    );
+
+    verify(redisTemplate).opsForValue();
+    verify(opsForValue)
+        .setIfAbsent(String.format("seriesSetHashes|%s|%s", tenant, seriesSetHash), "");
+
+    verifyNoMoreInteractions(cqlTemplate, cassandraTemplate, redisTemplate, opsForValue);
+  }
+
+  @Test
+  void stopAtRedisWhenPresentThere() {
+    when(redisTemplate.opsForValue())
+        .thenReturn(opsForValue);
+    when(opsForValue.setIfAbsent(any(), any()))
+        // this time redis says the key was present
+        .thenReturn(Mono.just(false));
+
+    final String tenant = randomAlphanumeric(10);
+    final String seriesSetHash = randomAlphanumeric(10);
+    final String metricName = randomAlphanumeric(10);
+    final String tagK = randomAlphanumeric(5);
+    final String tagV = randomAlphanumeric(5);
+    final Map<String, String> tags = Map.of(tagK, tagV);
+
+    Mono.from(
+        metadataService.storeMetadata(tenant, seriesSetHash, metricName, tags)
+    ).block();
+
+    verify(redisTemplate).opsForValue();
+    verify(opsForValue)
+        .setIfAbsent(String.format("seriesSetHashes|%s|%s", tenant, seriesSetHash), "");
+
+    // store the same again, but it should hit cache and not redis
+    Mono.from(
+        metadataService.storeMetadata(tenant, seriesSetHash, metricName, tags)
+    ).block();
+
+    verifyNoMoreInteractions(cqlTemplate, cassandraTemplate, redisTemplate, opsForValue);
+  }
+
+  @Test
+  void confirmMaxCacheConfig() {
```

Comment:
Is this the right name for the test?  I'm not understanding how this "confirms max"?

Reply:
It's named for what I meant to do in the test, but looking at the test again I think I missed a step I wanted to verify. The inline comment is also wrong. I'll fix these.

Reply:
I think I was viewing the only commit.  Looking at it now it all makes sense.

I had used the `View Changes` button to only see the latest changes since my last review and for some reason all this new code wasn't there.

...now I'm not sure what else I've missed.  Will do a quick scan of the rest.